src/Route.php

```diff
@@ -11,15 +11,21 @@ class Route
 
     public string $regex;
 
-    /** @var mixed[] */
+    /** @var array<string, string> */
     public array $variables;
 
     /** @var mixed */
     public $handler;
 
+    /** @param array{httpMethod: string, handler: mixed, regex: string, variables: array<string, string>} $state */
+    public static function __set_state(array $state): self
```

Comment:
Unfortunately the difference is larger than that: Previously, the route data was a static array, which means that the whole result is computed once during compilation and then stored as an immutable value in shared memory. In this case, loading the data is completely free.

Introducing any kind of non-static value removes that optimization -- it is now necessary to execute the file and create *all* Route objects at runtime (even those routes that will not be used by the current request).

This is not visible in a typical micro-benchmark setup, because everything runs in one "request" there, but I would expect that this makes a visible difference when tested against a web server (with opcache).

Reply:
That makes sense, thanks! I'll reorganise things to keep the old behaviour but still try to improve some things.